src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -613,7 +629,10 @@ protected static bool TryLookupConstructorParameter(
                 createExtensionProperty: false);
 
             // Mark the property as read from the payload if required.
-            state.Current.MarkRequiredPropertyAsRead(jsonPropertyInfo);
+            if (!useExtensionProperty)
```

Comment:
Why is the method guarded by this boolean now? I would assume we validate required properties even if the type uses an extension property.